### PR TITLE
Invite settings config

### PR DESCRIPTION
**Pull request by tdaga**
Allows users to enable/disable racing invite settings. 
invite settings

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -195,6 +204,12 @@ function appStateReducer(state: CoveyAppState, update: CoveyAppUpdate): CoveyApp
         nextState.nearbyPlayers = state.nearbyPlayers;
       }
       break;
+    case 'playerRaceSettings':
```
Comment (kaseyl) [inline frontend/src/App.tsx]: Maybe this should be called 'updatePlayerRaceSettings', and the other one as 'toggleRaceSettings' instead.  The name's a little confusing

**Review by kaseyl — COMMENTED** [review]
I totally thought we were gonna take the easy way out (and just have it binded to the backend) but this looks so much better. Nice!
Just one small naming comment

File: frontend/src/components/VideoCall/VideoFrontend/components/MenuBar/MenuBar.tsx [hunk 2]
```diff
@@ -82,6 +84,21 @@ export default function MenuBar(props: { setMediaError?(error: Error): void }) {
             <Grid container justify="center">
               <ToggleAudioButton disabled={isReconnecting} setMediaError={props.setMediaError} />
               <ToggleVideoButton disabled={isReconnecting} setMediaError={props.setMediaError} />
+              <FormControlLabel
+                control={
+                <Switch
+                  checked={appState.enableInvite}
+                  onChange={
+                    event => {
+                    event.stopPropagation();
```
Comment (jean-zhang) [inline frontend/src/components/VideoCall/VideoFrontend/components/MenuBar/MenuBar.tsx]: What are we stopping propagation for?

File: frontend/src/classes/Player.ts [hunk 3]
```diff
@@ -9,10 +9,15 @@ export default class Player {
...
   public label?: Phaser.GameObjects.Text;
 
-  constructor(id: string, userName: string, location: UserLocation) {
+  private _enableInvite?: boolean;
+
+  constructor(id: string, userName: string, location: UserLocation, enableInvite?: boolean) {
     this._id = id;
     this._userName = userName;
     this.location = location;
+    if(enableInvite !== null) {
```
Comment (jean-zhang) [inline frontend/src/classes/Player.ts]: run formatting - missing space

File: frontend/src/components/VideoCall/VideoFrontend/components/ParticipantInfo/ParticipantInfo.tsx [hunk 4]
```diff
@@ -281,6 +281,7 @@ export default function ParticipantInfo({
                   }
                 }
               }}
+              disabled={ !(appState.players.find(player => player.id === profile?.id)?.enableInvite) }
```
Comment (jean-zhang) [inline frontend/src/components/VideoCall/VideoFrontend/components/ParticipantInfo/ParticipantInfo.tsx]: run formatting

File: services/roomService/src/types/CoveyTownListener.ts [hunk 5]
```diff
@@ -47,6 +47,13 @@ export default interface CoveyTownListener {
   onMazeGameResponded(senderPlayer: Player, recipientPlayer: Player, gameAcceptance: boolean): void;
 
   /**
+   * Called when a player updates their racing settings
+   * @param recipientPlayer the player that has been invited
```
Comment (jean-zhang) [inline services/roomService/src/types/CoveyTownListener.ts]: params incorrect

File: frontend/src/components/VideoCall/VideoFrontend/components/MenuBar/MenuBar.tsx [hunk 6]
```diff
@@ -82,6 +84,17 @@ export default function MenuBar(props: { setMediaError?(error: Error): void }) {
             <Grid container justify="center">
               <ToggleAudioButton disabled={isReconnecting} setMediaError={props.setMediaError} />
               <ToggleVideoButton disabled={isReconnecting} setMediaError={props.setMediaError} />
+              <FormControlLabel
+                control={
+                  <Switch
+                    checked={appState.enableInvite}
+                    onChange={() => { appState.emitRaceSettings(appState.myPlayerID, appState.enableInvite); }}
```
Comment (jean-zhang) [inline frontend/src/components/VideoCall/VideoFrontend/components/MenuBar/MenuBar.tsx]: run formatting on this file should be able to be simplified to

`onChange={() => appState.emitRaceSettings(appState.myPlayerID, appState.enableInvite)}`

**Pull request merged**